tutorverse-server/server.js

```diff
@@ -285,15 +284,37 @@ app.get("/chatlist", async (req,res) => {
       return res.status(404).json({ message: error, list: null });
     }
   } else{
-    Message.findAll({ where: {recepientId: req.session.user.id}}).then(messages => {
-      const studentIDs = messages.map(message => message.senderId);
-      User.findAll({where: {id: studentIDs}})
-      .then(students => res.json(students))
-      .catch(error => {
-        console.error('Error fetching students:', error);
-        res.status(500).json({ error: "Internal server error"})
-      })
+    const studentIDs = new Set();
+    
+    Message.findAll({ 
+      where: {recepientId: req.session.user.id},
+      order: [['createdAt', 'DESC']]
+    
+    }).then(messages => {
+      messages.map((message) => 
+        studentIDs.add(message.senderId)
+      );
```

Comment:
Check out the mdn docs for array.map. It actually creates and return a new array based on what you return from the callback.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

So here, we actually could do something like

```javascript
const studentIDs = messages.map(message => {
  return message.senderId;
});
```